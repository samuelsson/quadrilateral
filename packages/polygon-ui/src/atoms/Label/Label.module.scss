.Label {
  display: block;
}

.Error {
  color: var(--color-red);
}

.Disabled {
  opacity: 0.5;
  cursor: not-allowed;
}
