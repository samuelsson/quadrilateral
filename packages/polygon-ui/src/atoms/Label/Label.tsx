import React from 'react';
import cn from 'classnames';
import styles from './Label.module.scss';

export interface LabelProps {
  text: string;
  htmlFor: string;
  error?: boolean;
  disabled?: boolean;
}

const Label = ({ htmlFor, text, error, disabled }: LabelProps): JSX.Element => (
  <label
    htmlFor={htmlFor}
    className={cn(
      styles.Label,
      error && styles.Error,
      disabled && styles.Disabled
    )}
  >
    {text}
  </label>
);

export default Label;
