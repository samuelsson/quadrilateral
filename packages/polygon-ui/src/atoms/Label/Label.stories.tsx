/* eslint-disable react/jsx-props-no-spreading */
import * as React from 'react';
import { Story, Meta } from '@storybook/react';
import { hideControls } from '../../helpers/storybookHelper';
import Label, { LabelProps } from './Label';

export default {
  component: Label,
  title: 'Atoms / Label',
  args: {
    text: 'Label',
  },
  argTypes: {
    ...hideControls(['htmlFor']),
  },
} as Meta;

export const Default: Story<LabelProps> = (args) => (
  <Label htmlFor="label" {...args} />
);
