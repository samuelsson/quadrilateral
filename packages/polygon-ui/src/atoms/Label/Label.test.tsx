import * as React from 'react';
import { render } from '@testing-library/react';
import Label from './Label';

describe('Label', (): void => {
  it('renders with no errors', (): void => {
    const { asFragment } = render(<Label htmlFor="input" text="Label Text" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
