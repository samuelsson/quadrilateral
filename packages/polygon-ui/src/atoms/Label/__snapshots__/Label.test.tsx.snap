// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders with no errors 1`] = `
<DocumentFragment>
  <label
    class="Label"
    for="input"
  >
    Label Text
  </label>
</DocumentFragment>
`;
